.github/workflows/e2e-leader-election.yml

```diff
@@ -0,0 +1,76 @@
+name: E2E Leader Election
+
+on:
+  pull_request:
```

Comment:
since it is running on Pull Requests, the time is takes to run matters a lot. It was the most time consuming check on your PR. Please, see if you can make it run as fast as `Contracts Rocks Tests` or `Relayer`

Reply:
I will reduce the amount of iterations to maintain below the relayer test